test/framework/interface.go

```diff
@@ -105,9 +105,11 @@ type Cluster interface {
 	CreateNamespace(namespace string) error
 	DeleteNamespace(namespace string) error
 	DeployApp(namespace, appname, token string) error
+	DeployExternalApp(namespace, appname, id string) error
```

Comment:
can we follow the new `deployment` pattern that I did with Tracing? I'd really like to avoid adding more and more methods to `Cluster` interface. I can pair program on this if you wish

Reply:
Looks great now, thanks for the pointer!